File: Algorithms/src/nattybear/level1/MakePrime.js
const solution = nums => {
  let answer = 0;
  const size = nums.length;
  for (let i = 0; i < size; i++) {
    for (let j = i+1; j < size; j++) {
      for (let k = j+1; k < size; k++) {
        let n = nums[i] + nums[j] + nums[k];
        if (isPrime(n)) answer++;
      }
    }
  }
  return answer;
}

const isPrime = num => {
  for (let i = 2, s = Math.sqrt(num); i <= s; i++)
    if (num % i === 0)
      return false;
  return num > 1;
}

console.log(solution([1,2,3,4]));
console.log(solution([1,2,7,6,4]));

File: Algorithms/src/nattybear/level1/MakePrime.java
package nattybear.level1;

class Solution {
  public int solution(int[] nums) {
    int answer = 0;
    int size = nums.length;
    for (int i = 0; i < size; i++) {
      for (int j = i+1; j < size; j++) {
        for (int k = j+1; k < size; k++) {
          int n = nums[i] + nums[j] + nums[k];
          if (isPrime(n)) answer++;
        }
      }
    }
    return answer;
  }

  boolean isPrime(int n) {
    double s = Math.sqrt(n);
    for (int i = 2; i <= s; i++)
      if (n % i == 0)
        return false;
    return n > 1;
  }
}

public class MakePrime {
  public static void main(String[] args) {
    Solution s = new Solution();

    int[] xs = {1, 2, 3, 4};
    int[] ys = {1, 2, 7, 6, 4};

    System.out.println(s.solution(xs));
    System.out.println(s.solution(ys));
  }
}
